### PR TITLE
fix: add callback test for missing payment in signed state

### PR DESCRIPTION
**Pull request by swapnil-turing**

### CODE REVIEW

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/checkout/views.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def get(self, request, *args, **kwargs):
                 max_age=60 * 60 * 24,
             )
             payment_id = payload["payment_id"]
-            payment = get_object_or_404(Payment, id=payment_id)
+            payment = Payment.objects.get(id=payment_id)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/views.py]: value:useful; category:bug; feedback:Selected 'useful' over 'delightful' because this is a high-severity correctness issue that could return 500 instead of 404, but it likely would have been caught in normal review when validating the new missing-payment test path.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/checkout/tests.py [hunk 2]
```diff
@@ -851,6 +851,14 @@ def test_callback_rejects_tampered_state(self, mock_update_status):
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
         mock_update_status.assert_not_called()
 
+    @patch("checkout.views.StripeService.check_and_update_payment_status")
+    def test_callback_rejects_signed_state_with_missing_payment(self, mock_update_status):
+        callback_state = signing.dumps({"payment_id": "00000000-0000-0000-0000-000000000000"}, salt="checkout.stripe.redirect")
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/tests.py]: value:good-to-have; category:bug; feedback:Selected 'good-to-have' over 'useful' because this improves test resilience against edge-case fixture collisions, but it is low severity and does not affect production behavior.

**Comment by swapnil-turing** [comment]
augment review

**Review by michaelos443 — APPROVED** [review]
